File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    // 결과를 저장할 배열 초기화
    let answer = [];

    // commands 배열을 순회하면서 각 명령에 대해 작업 수행
    for (let c of commands) {
        // 배열에서 c[0]번째부터 c[1]번째까지 잘라낸다
        // JavaScript에서 slice는 시작 인덱스를 포함하고, 종료 인덱스는 포함하지 않으므로
        // c[0] - 1부터 c[1]까지 지정해야 정확한 구간을 추출할 수 있다
        let slice = array.slice(c[0] - 1, c[1]);

        // 잘라낸 배열을 숫자 오름차순으로 정렬
        // (a, b) => a - b를 써야 숫자 기준으로 정확하게 정렬된다
        slice.sort((a, b) => a - b);

        // 정렬된 배열에서 c[2]번째 요소를 꺼낸다
        // 배열 인덱스는 0부터 시작하므로 c[2] - 1로 접근해야 함
        answer.push(slice[c[2] - 1]);
    }

    // 모든 명령에 대한 결과가 저장된 배열을 반환
    return answer;
}

// ✅ 예제 테스트 실행
console.log(solution(
    [1, 5, 2, 6, 3, 7, 4],
    [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
));
// 출력 결과: [5, 6, 3]

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.ArrayList;
import java.util.List;
import java.util.Arrays;

public class Solution { 
    public List<Integer> solution(int[] array, int[][] commands) { 
        // 결과를 저장할 리스트 초기화
        List<Integer> answer = new ArrayList<>(); 

        // commands 배열을 순회하면서 각 연산 수행
        for (int[] c : commands) { 
            // 배열의 i번째부터 j번째까지 슬라이싱 (배열 인덱스는 0부터 시작하므로 c[0] - 1 사용)
            int[] slice = Arrays.copyOfRange(array, c[0] - 1, c[1]); 
            
            // 잘라낸 배열을 정렬
            Arrays.sort(slice); 
            
            // 정렬된 배열에서 k번째 요소를 결과 리스트에 추가 (배열 인덱스는 0부터 시작하므로 c[2] - 1 사용)
            answer.add(slice[c[2] - 1]); 
        }

        // 최종 결과 반환
        return answer;
    }

    public static void main(String[] args) { 
        // 예제 테스트 실행
        Solution sol = new Solution(); 
        System.out.println(sol.solution(
            new int[]{1, 5, 2, 6, 3, 7, 4}, 
            new int[][]{{2, 5, 3}, {4, 4, 1}, {1, 7, 3}}
        ));
        // 출력 결과: [5, 6, 3]
    }
}
